ocaml/typing/typedecl.ml

```diff
@@ -1098,6 +1098,14 @@ let check_coherence env loc dpath decl =
   match decl with
     { type_kind = (Type_variant _ | Type_record _| Type_open);
       type_manifest = Some ty } ->
+      if !Clflags.allow_illegal_crossing then begin 
```

Comment:
What's going on here? This doesn't seem to be related to this patch. I think I'm missing something.

Reply:
This prevents illegal crossing for types that have both a manifest and a non-abstract kind. For example, without this, the following is accepted:
```
type a = { foo : string }
type b : value mod portable = a = { foo : string }
```
My understanding is that this should not be accepted.